core/local/stater.js

```diff
@@ -0,0 +1,77 @@
+/* @flow */
+
+const fse = require('fs-extra')
+
+let winfs
+if (process.platform === 'win32') {
+  // $FlowFixMe
+  winfs = require('@gyselroth/windows-fsstat')
+}
+
+/*::
+import type { Metadata } from '../metadata'
+import type { Callback } from '../utils/func'
+import type fs from 'fs'
+
+export type WinStats = {
+  fileid: string,
+  ino: number,
+  size: number,
+  atime: Date,
+  mtime: Date,
+  ctime: Date,
+  directory: bool,
+  symbolicLink: bool
+}
+export type Stats = fs.Stats | WinStats
+*/
+
+module.exports = {
+  stat: async function (filepath /*: string */) {
+    if (!winfs) {
+      return fse.stat(filepath)
+    }
+    return new Promise((resolve, reject) => {
+      try {
+        // XXX It would be better to avoid sync IO operations, but
+        // before node 10.5.0, it's our only choice for reliable fileIDs.
+        resolve(winfs.lstatSync(filepath))
+      } catch (err) {
+        reject(err)
+      }
+    })
+  },
+
+  withStats: function (filepath /*: string */, callback /*: Callback */) {
```

Comment:
Is there a reason why you're using the `property: function` declaration style instead of the new style used below for `isDirectory` and such?

Reply:
yes, the goal is to avoid rewriting https://github.com/cozy-labs/cozy-desktop/blob/master/core/local/index.js#L180-L278 which use callbacks (this PR seems big enough without that).